checker/checker.go

```diff
@@ -184,6 +190,14 @@ func (c *Checker) lookForSeenCerts(ctx context.Context, crl *x509.RevocationList
 	var seenSerials [][]byte
 	for _, seen := range crl.RevokedCertificateEntries {
 		if metadata, ok := unseenCerts[db.NewCertKey(seen.SerialNumber).SerialString()]; ok {
+			idp, err := getIDP(crl)
+			if err != nil {
+				return err
+			}
+			if metadata.CRLDistributionPoint != "" && metadata.CRLDistributionPoint != idp {
+				return fmt.Errorf("cert %x on CRL %q has CRLDistributionPoint %q",
+					seen.SerialNumber, idp, metadata.CRLDistributionPoint)
```

Comment:
If we return with an error here, we won't delete them below, and we'd get an ever-growing set of "unseen" certificates in dynamo.

We could accumulate a set of errors in this loop, then errors.Join them together for return after deleting.

Reply:
Good call.